metadata-ingestion/src/datahub/ingestion/source/sagemaker_processors/models.py

```diff
@@ -7,16 +7,37 @@
 from datahub.ingestion.api.workunit import MetadataWorkUnit
 from datahub.ingestion.source.sagemaker_processors.common import SagemakerSourceReport
 from datahub.ingestion.source.sagemaker_processors.jobs import JobDirection, ModelJob
-from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import MLModelSnapshot
+from datahub.ingestion.source.sagemaker_processors.lineage import LineageInfo
+from datahub.metadata.com.linkedin.pegasus2avro.metadata.snapshot import (
+    MLModelEndpointSnapshot,
+    MLModelSnapshot,
+)
 from datahub.metadata.com.linkedin.pegasus2avro.mxe import MetadataChangeEvent
-from datahub.metadata.schema_classes import MLModelPropertiesClass
+from datahub.metadata.schema_classes import (
+    EndpointStatusClass,
+    MLModelEndpointPropertiesClass,
+    MLModelPropertiesClass,
+)
+
+ENDPOINT_STATUS_MAP: Dict[str, str] = {
+    "OutOfService": EndpointStatusClass.OUT_OF_SERVICE,
+    "Creating": EndpointStatusClass.CREATING,
+    "Updating": EndpointStatusClass.UPDATING,
+    "SystemUpdating": EndpointStatusClass.UPDATING,
+    "RollingBack": EndpointStatusClass.ROLLING_BACK,
+    "InService": EndpointStatusClass.IN_SERVICE,
+    "Deleting": EndpointStatusClass.DELETING,
+    "Failed": EndpointStatusClass.FAILED,
+    "Unknown": EndpointStatusClass.UNKNOWN,
+}
```

Comment:
do we need this level of granularity in our model as well?

Reply:
I think so – it may be a little too detailed for our current pull-based integration, but it will be useful once we have push-based functionality.